src/webui/api/torrentscontroller.cpp

```diff
@@ -1041,3 +1041,70 @@ void TorrentsController::categoriesAction()
 
     setResult(categories);
 }
+
+void TorrentsController::addTagsAction()
+{
+    checkParams({"hashes", "tags"});
+
+    const QStringList hashes {params()["hashes"].split('|')};
+    const QStringList tags {params()["tags"].split(',', QString::SkipEmptyParts)};
+
+    for (const QString &tag : tags) {
+        const QString tagTrimmed {tag.trimmed()};
```

Comment:
This is favorite way of @Chocobo1 - to produce additional (const) variables. I would just make `tag` changeable:
```c++
for (QString tag : tags) {
        tag = tag.trimmed();
```
However, I do not want to discuss this topic, so that it does not become an unnecessary obstacle here.

Another question (also not for this PR) is that I do not think "trimming" (frequently used in this API) is correct.
For example:
1. if a tag with spaces at the edges is considered correct, why do we trim it?
2. otherwise, if a tag with spaces at the edges is considered incorrect, why we don't reject this "bad request"?

Reply:
>This is favorite way of @Chocobo1 - to produce additional (const) variables. 

Not so precise, because it is clear what are the critical values in a piece of code.

>I would just make tag changeable:

You don't even need to make it mutable! such as:
```c++
for (const QString &tag : tags) {
    applyToTorrents(hashes, [&tag](BitTorrent::TorrentHandle *const torrent)
    {
        torrent->addTag(tag.trimmed());
    });
}
```